pkg/controller/operators/olm/apiservices.go

```diff
@@ -164,60 +162,27 @@ func (a *Operator) checkAPIServiceResources(csv *v1alpha1.ClusterServiceVersion,
 		if serviceAccountName == "" {
 			serviceAccountName = "default"
 		}
-		serviceAccount, err := a.lister.CoreV1().ServiceAccountLister().ServiceAccounts(deployment.GetNamespace()).Get(serviceAccountName)
+		_, err = a.opClient.KubernetesInterface().CoreV1().ServiceAccounts(deployment.GetNamespace()).Get(context.TODO(), serviceAccountName, metav1.GetOptions{})
 		if err != nil {
-			logger.WithField("serviceaccount", serviceAccountName).Warnf("could not retrieve ServiceAccount")
+			logger.WithError(err).WithField("serviceaccount", serviceAccountName).Warnf("could not retrieve ServiceAccount")
 			errs = append(errs, err)
-			continue
 		}
 
-		// Ensure RBAC permissions for the APIService are correct
```

Comment:
Note ... the changes here have us no longer checking for the (Cluster)Role in `kube-system` namespace. I don't see what the point of that logic is - there's no remediation if it's missing, and any conformant Kube installation will have them. If we _really_ think we need to check for those, we can use a live client.

Reply:
Looking at this code, it seems like the idea was that:
- k8s comes with some predefined roles that are generally applicable to apiservices?
- instead of hardcoding the rules from those roles that we expect our serviceaccount to have, we can lookup those rules (in case they change from one k8s version to the next)
- then once we've compiled all the rules we expect our service account to have, run it through the rule checker, which doesn't actually care which roles/bindings made it happen.

And the new code is: "assume RBAC (Cluster)Roles contain the right rules (either OLM created it, or it is present in a conformant cluster); assume bindings bind the service account to the right (Cluster)Roles (OLM created it); now just look to see if the RBAC objects exist"

Am I getting that right? If so, I agree with you that the new code seems sufficient.